Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs

```diff
@@ -16,27 +17,98 @@ public partial class TMToastContents : PopupPage
 
     #region Public Properties
 
-    public ImageSource LeftIconSource { get; set; }
-
     public string Message { get; set; }
 
     public string RightIconText { get; set; }
 
     public double ToastWidthRequest { get; set; }
-
-    public Color ToastBackground { get; set; }
-
-    public Color TextColor { get; set; }
-
-    #endregion
-
+
+    #endregion
+
+    public static readonly BindableProperty ToastBackgroundColorProperty =
+        BindableProperty.Create(nameof(ToastBackgroundColor), typeof(Color), typeof(TMToast), Colors.Black);
+
+    public static readonly BindableProperty TextColorProperty =
+        BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(TMToast), Colors.White);
+
+    public static readonly BindableProperty LeftIconTintColorProperty =
+        BindableProperty.Create(nameof(LeftIconTintColor), typeof(Color), typeof(TMToast), Colors.White,
+            propertyChanged: OnLeftIconTintColorChanged);
+
+    public static readonly BindableProperty RightIconTintColorProperty =
+    BindableProperty.Create(nameof(RightIconTintColor), typeof(Color), typeof(TMToast), Colors.White,
+        propertyChanged: OnRightIconTintColorChanged);
+
+
     internal TMToastContents(string message, string actionButtonText, ToastTheme theme, Action action, bool isDismissable)
     {
         InitializeComponent();
-        SetTheme(theme.ToString());
+        SetTheme(theme.ToString());
+        SetDynamicResource(StyleProperty, theme.ToString());
         PopupData(message, actionButtonText, action, isDismissable);
         BindingContext = this;
         CloseAfterDelay();
+    }
+    public Color ToastBackgroundColor
```

Comment:
Can't we have them as internal?

Reply:
As the binding is implemented in xaml, it has to be public.